### PR TITLE
Feat/shards backend integration

### PR DESCRIPTION
**Pull request by osmaczko**
### What does the PR do

closes: #12408

status-go: https://github.com/status-im/status-go/pull/4256/

### Affected areas

- communities

### Screenshot of functionality (including design for comparison)

https://github.com/status-im/status-desktop/assets/33099791/b2911dff-7aac-42b8-b016-3e5d4af8418d

### CODE REVIEW

**Comment by osmaczko** [comment]
@benjthayer @caybro I've changed the flow a bit in comparison to designs, please take a look:

https://github.com/status-im/status-desktop/assets/33099791/23e58dfe-a554-4476-841e-fe2e412c097d

The reason for this change is that the updated pubsubTopic becomes available only after the sharding has been configured.

**Comment by benjthayer** [comment]
> @benjthayer @caybro I've changed the flow a bit in comparison to designs, please take a look:
> 
>  community-sharding-storybook-2023-11-02_22.44.11.mp4 
> The reason for this change is that the updated pubsubTopic and the key becomes available only after the sharding has been configured.

No problem, thanks for letting me know @osmaczko - I will adjust the designs to reflect cc @John-44

File: ui/app/AppLayouts/Communities/panels/EditSettingsPanel.qml [hunk 1]
```diff
@@ -111,30 +117,32 @@ StatusScrollView {
...
-                    console.warn("TODO: enable community sharding for shardIndex:", shardIndex) // TODO community sharding backend
-                    root.communityShardIndex = shardIndex
+                shardIndex: root.shardIndex
+                pubsubTopic: '{"pubsubTopic":"%1", "publicKey":"%2"}'.arg(root.pubsubTopic).arg(root.pubsubTopicKey)
+                shardingInProgress: root.shardingInProgress
+
+                onShardIndexChanged: root.shardIndexEdited(shardIndex)
+                onShardingInProgressChanged: if (!shardingInProgress) {
+                    // bring back the binding
+                    enableShardingPopup.shardIndex = Qt.Binding(() => root.shardIndex)
```
Comment (caybro) [inline ui/app/AppLayouts/Communities/panels/EditSettingsPanel.qml:130]:
```suggestion
                    enableShardingPopup.shardIndex = Qt.binding(() => root.shardIndex)
```

Typo?

**Review by caybro — APPROVED** [review]
LGTM

Just one suspicious typo

File: ui/app/AppLayouts/Communities/panels/EditSettingsPanel.qml (hunk 1 above)
Comment (osmaczko) [reply ui/app/AppLayouts/Communities/panels/EditSettingsPanel.qml]: Yes, good catch, thanks.

**Pull request merged**